addons/account/models/account_move_line.py

```diff
@@ -1541,6 +1541,25 @@ def copy_data(self, default=None):
                 line._copy_data_extend_business_fields(values)
         return data_list
 
+    def _search_panel_domain_image(self, field_name, domain, set_count=False, limit=False):
+        if field_name != 'account_root_id' or set_count:
+            return super()._search_panel_domain_image(field_name, domain, set_count, limit)
+
+        # Override in order to not read the complete move line table and use the index instead
+        query = self._search(domain, limit=1)
+        query.order = None
+        query.add_where('account.id = account_move_line.account_id')
+        query_str, query_param = query.select()
+        self.env.cr.execute(f"""
+            SELECT account.root_id
+              FROM account_account account,
+                   LATERAL ({query_str}) line
+        """, query_param)
+        return {
+            root.id: {'id': root.id, 'display_name': root.display_name}
+            for root in self.env['account.root'].browse(id for [id] in self.env.cr.fetchall())
```

Comment:
while we're at it, wouldn't that speed up to avoid browsing and computing the display name of accounts by reading that in the SQL query directly? 

I'm kinda hesitant on that: it would be better for consistency and robustness to use computed display_name as you do, _but_ if the perfs are suffering I'd happily pay that price. So, we need to test and 
benchmark that change :timer_clock:

Reply:
There is no real benchmark to do, this part depends on the number of accounts, which is the same as on runbot instances, where it is already instant.
We need to browse and fetch them anyways in this same transaction because we need to compute the tree via the `parent_id` field, so everything is cached.